processor/cumulativetodeltaprocessor/internal/tracking/tracker.go

```diff
@@ -81,19 +123,45 @@ func (t *MetricTracker) Convert(in MetricPoint) (out DeltaValue, valid bool) {
 	hashableID := b.String()
 	identityBufferPool.Put(b)
 
+	var keep bool
 	s, ok := t.states.LoadOrStore(hashableID, &State{
 		PrevPoint: metricPoint,
 	})
 	if !ok {
-		return
+		switch t.initialValue {
+		case InitialValueDrop:
+			return
+		case InitialValueAuto:
+			if metricID.StartTimestamp < t.startTime || metricPoint.ObservedTimestamp == metricID.StartTimestamp {
+				return
+			}
+			out.StartTimestamp = metricID.StartTimestamp
+			keep = true
+		case InitialValueKeep:
+			keep = true
+		}
 	}
+
 	valid = true
 
+	if keep {
+		switch metricID.MetricType {
+		case pmetric.MetricTypeHistogram:
+			*out.HistogramValue = metricPoint.HistogramValue.Clone()
+		case pmetric.MetricTypeSum:
+			out.IntValue = metricPoint.IntValue
+			out.FloatValue = metricPoint.FloatValue
+		}
+		return
+	}
+
 	state := s.(*State)
 	state.Lock()
 	defer state.Unlock()
 
-	out.StartTimestamp = state.PrevPoint.ObservedTimestamp
+	if ok {
```

Comment:
I don't believe it is possible to reach this condition unless `ok` is `true`.

Reply:
ah right, done